### DISCUSSION TITLE
Bin growth by individual axis

### DISCUSSION THREAD

**Discussion by gstankov** [Padam87/bin-packer#8] [General]
Hey,

using your package in a side project for nesting PDF-s. Nice work :)

Is it possible to grow bin along a single axis? Would be nice for roll media types.

Also is there a way to check if node was rotated.  Just like '$block->isUsed()' method.

Thanks again, br,

**Comment by Padam87** [comment]
Hi,

> nesting PDF

Oh, like print plates? Sounds cool.


> Is it possible to grow bin along a single axis?

Not right now, but should be pretty easy to add. I don't plan on continuing this version of the packer, because I did a complete rewrite with heuristics, and that is much more flexible, and can give multiple choices. https://github.com/Padam87/bin-packer/pull/5

The reason it is not merged yet, is that I didn't have the time to add growth at all yet, but when I do it I will take this into account.

> Also is there a way to check if node was rotated.

Not ATM, but that is really easy to do, good for a PR if you want to create one.
https://github.com/Padam87/bin-packer/blob/master/src/Model/Block.php#L98

I'd add a rotated property (false by default), and just negate it any time ->rotate() is called.

**Comment by gstankov** [comment]
Hey,

thanks for merging my pull request and for your kind guidance.

My first contribution ever! :)

Here's outputs of a multipage PDF file nested with Bin-Packer. Each page is multiplied 35 times and placed on a 'sheet':

[Nest-Cut-64f89e1d51bb5.pdf](https://github.com/Padam87/bin-packer/files/12539817/Nest-Cut-64f89e1d51bb5.pdf) - Cut file
[Nest-Print-64f89e1d51bb5.pdf](https://github.com/Padam87/bin-packer/files/12539818/Nest-Print-64f89e1d51bb5.pdf) - Print file

Original file: 
[Shapes - 35kom.pdf](https://github.com/Padam87/bin-packer/files/12540003/Shapes.-.35kom.pdf)

It's a common 'Print-Cut' workflow in digital printing. One file you print and the other one you use in cutting software. I have it currently as a private repo but will make it public someday. Currently it's customized for one brand of vinyl cutter only and not quite polished.

So there it is, Bin-Packer in action :)

Thanks!

**Comment by Padam87** [comment]
Ah, I see... for digital this should be okay, but for offset it will get much more complicated :)
